great_expectations/expectations/metrics/multicolumn_map_metrics/select_column_values_unique_within_record.py

```diff
@@ -16,9 +24,37 @@ class SelectColumnValuesUniqueWithinRecord(MulticolumnMapMetricProvider):
         "ignore_row_if",
     )
 
-    # TODO: <Alex>ALEX -- temporarily only a Pandas implementation is provided (others to follow).</Alex>
+    # TODO: <Alex>ALEX -- temporarily only Pandas and SQL Alchemy implementations are provided (Spark to follow).</Alex>
     @multicolumn_condition_partial(engine=PandasExecutionEngine)
     def _pandas(cls, column_list, **kwargs):
         num_columns = len(column_list.columns)
         row_wise_cond = column_list.nunique(dropna=False, axis=1) >= num_columns
         return row_wise_cond
+
+    @multicolumn_condition_partial(engine=SqlAlchemyExecutionEngine)
+    def _sqlalchemy(cls, column_list, **kwargs):
+        """
+        NOTE: The present approach relies on an inefficient query condition construction implementation, whose
+        computational cost is O(num_columns^2).  However, until a more efficient implementation compatible with
+        SQLAlchemy is available, this is the only feasible mechanism under the current architecture, where map metric
+        providers must return a condition (and do not have access to the selectable, such as the table of Batch data).
+        Nevertheless, query length limit for SQL databases is 1GB, and database engines processing queries efficiently.
+        """
+        num_columns = len(column_list)
+
+        if num_columns > 100:
+            logger.warning(
+                f"""Batch data with {num_columns} columns is detected.  Computing the "{cls.condition_metric_name}" \
+metric for wide tables using SQLAlchemy leads to long WHERE clauses for the underlying database engine to process.
+"""
+            )
```

Comment:
Did you test/compute if 100 columns is where performance suffers or is that just a round number?

Reply:
@NathanFarmer Just a round number for a warning.  It should all work fine, because of the 1GB query size limit.